nomad/job_endpoint_hooks_test.go

```diff
@@ -1280,6 +1280,51 @@ func Test_jobImpliedConstraints_Mutate(t *testing.T) {
 			expectedOutputWarnings: nil,
 			expectedOutputError:    nil,
 		},
+		{
+			name: "task with secret",
+			inputJob: &structs.Job{
+				Name: "example",
+				TaskGroups: []*structs.TaskGroup{
```

Comment:
Having coverage of the multiple group/task scenario described above would be nice.